ui/components/src/Table/Table.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  useReactTable,
+  getCoreRowModel,
+  ColumnDef,
+  RowSelectionState,
+  OnChangeFn,
+  CoreOptions,
+} from '@tanstack/react-table';
+import { useTheme } from '@mui/material';
+import { useCallback, useMemo } from 'react';
+import { VirtualizedTable } from './VirtualizedTable';
+import { TableCheckbox } from './TableCheckbox';
+import { TableProps, persesColumnsToTanstackColumns } from './model/table-model';
+
+/**
+ * Component used to render tabular data in Perses use cases. This component is
+ * **not** intended to be a general use data table for use cases unrelated to Perses.
+ *
+ * **Note: This component is currently experimental and is likely to have significant breaking changes in the near future. Use with caution outside of the core Perses codebase.**
```

Comment:
Let me know if folks have other thoughts on how to word this. The important thing to me is to:

1. Communicate that we are NOT trying to provide a general use data table to compete with things like MUI DataGrid or the like. I don't think we want to be in the business of supporting that.
2. Communicate that the table is currently in flux and should be used with caution. We should probably leave this on here until the table legend and table panel feature are completed.